sdks/go/test/regression/coders/fromyaml/fromyaml.go

```diff
@@ -218,6 +218,19 @@ func diff(c Coder, elem *exec.FullValue, eg yaml.MapItem) bool {
 		}
 		return pass
 
+	case "beam:coder:nullable:v1":
+		if elem.Elm == nil || eg.Value == nil {
+			got, want = elem.Elm, eg.Value
+		} else {
+			got = string(elem.Elm.([]byte))
```

Comment:
This code has design problems, but they pre-exist John's addition, so he's largely following the existing structure. In this case, this was an unscheduled task that I had to do in a hurry, with little oversight, and improving it has never been a priority. This can certainly be improved significantly, but likely in a separate PR.

So generally, I agree with Danny here. But specifically...

1. As this is specific purpose comparison code, it is true it may blow up, but it will only blow up with a runtime type assertion error when someone is adding new/strange cases to it. The [defer'd recover](https://github.com/apache/beam/blob/488e1d165e121d757fac8b9260d004de93dbd338/sdks/go/test/regression/coders/fromyaml/fromyaml.go#L122) at the top of the encoder case loop will narrow down where the problem is, if not the actual example that blew up, which should be enough information to determine the offending types. 

2. Agreed, in this case, there's probably a separate bug going on. But this is how the plain `bytes` coder is handling it. IIRC there was something very odd with the YAML parsing which lead to flaky interpretations of the types. It was easier to work around it at the time.

3. In this case, I disagree: Downstream the `want` and `got` are compared with `cmp`, and that will produce a much clearer error message than what we could add here, along with the full coder information.